lib/srv/regular/sshserver.go

```diff
@@ -765,8 +764,9 @@ func (s *Server) serveAgent(ctx *srv.ServerContext) error {
 		return trace.ConvertSystemError(err)
 	}
 
-	// start an agent on a unix socket
-	agentServer := &teleagent.AgentServer{Agent: ctx.Parent.GetAgent()}
+	// start an agent server on a unix socket.  each incoming connection
+	// will result in a separate agent request.
+	agentServer := &teleagent.AgentServer{Getter: teleagent.AgentGetter(ctx.Parent.StartAgentChannel)}
```

Comment:
I know you just copied over code, but we should use a function here, like `teleagent.New(...)`. It's not clear which/if the other fields need to be set.